### PR TITLE
Fix/appgw backend hostname

### PR DESCRIPTION
**Pull request by djackson-ans**
# Description

A couple of fixes & improvement to the App Gateway module including:

- Check to make sure the user is not using Lighthouse. Keyvault won't deploy with lighthouse.
- Changed WAF to a bool. Removed ability to use v1 SKU
- Made it so that `pick_host_name_from_backend_address` is automatically set when `host_name` is unset
- Made it so that `pick_host_name_from_backend_http_settings` is automatically set when `host` is unset

## Type of change

Please select the option that is relevant.

  - [X] Bug fix (non-breaking change which fixes an issue)
  - [X] New feature (non-breaking change which adds functionality)

# Checklist:

  - [X] My code follows the style guidelines of this project
  - [X] I have performed a self-review of my code
  - [X] I have commented my code, particularly in hard-to-understand areas
  - [X] I have made corresponding changes to the documentation
  - [X] My changes generate no new warnings e.g. code analysis tooling or general use of the changed code

### CODE REVIEW

File: azure/application-gateway/advanced/keyvault.tf [hunk 1]
```diff
@@ -11,6 +11,12 @@ resource "azurerm_key_vault" "main" {
...
   tenant_id                  = data.azurerm_client_config.current.tenant_id
+  lifecycle {
+    precondition {
+      condition     = data.azurerm_client_config.current.tenant_id == data.azurerm_subscription.current.tenant_id
+      error_message = "Your authenticated tenant does not match the tenant where you want to deploy. Please ensure that you are at least a guest user in the tenant where you want to deploy AppGW"
+    }
+  }
 }
 
 resource "azurerm_key_vault_access_policy" "main_user" {
```
Comment (jk-ans) [inline azure/application-gateway/advanced/keyvault.tf:22]: Does the identity need all those permissions.  e.g - Backup, Purge,

**Comment by jk-ans** [comment]
Looks good :)

**Review by jk-ans — APPROVED** [review]

File: azure/application-gateway/advanced/keyvault.tf (hunk 1 above)
Comment (djackson-ans) [reply azure/application-gateway/advanced/keyvault.tf:22]: Probably not, outside of the scope of the original PR but good spot regardless. Any suggestions on the permissions I could use here?

Comment (jk-ans) [reply azure/application-gateway/advanced/keyvault.tf:22]: Correct me if I'm wrong but it looks like only a certificate is being created.  So I'd suggest the following:
certificate_permissions = ["Create", "Get", "List", "Update", "Recover, "Restore"]

Comment (djackson-ans) [reply azure/application-gateway/advanced/keyvault.tf:22]: I've reviewed this further this morning and it seems like it also needs "Delete" to run terraform destroy

**Pull request merged**